src/EnergyPlus/UnitarySystem.cc

```diff
@@ -10747,35 +10757,44 @@ namespace UnitarySystems {
                     state.dataUnitarySystems->CompOnMassFlow = this->MaxNoCoolHeatAirMassFlow;
                     state.dataUnitarySystems->CompOnFlowRatio = this->m_NoLoadAirFlowRateRatio;
                     state.dataUnitarySystems->OACompOnMassFlow = this->m_NoCoolHeatOutAirMassFlow;
-                } else if (CoolSpeedNum == 1) {
-                    state.dataUnitarySystems->CompOnMassFlow = this->m_CoolMassFlowRate[1];
-                    state.dataUnitarySystems->CompOnFlowRatio = this->m_MSCoolingSpeedRatio[1];
+                } else if (this->m_EconoSpeedNum > 0) { // multi-stage economizer operation; set system flow rate to economizer flow rate
+                    state.dataUnitarySystems->CompOnMassFlow = this->m_CoolMassFlowRate[this->m_EconoSpeedNum];
+                    state.dataUnitarySystems->CompOnFlowRatio = this->m_MSCoolingSpeedRatio[this->m_EconoSpeedNum];
                     state.dataUnitarySystems->OACompOnMassFlow = this->m_CoolOutAirMassFlow;
-                } else if (CoolSpeedNum > 1) {
+                } else if (CoolSpeedNum > 0) {
```

Comment:
This logical stack is interesting. I thought the intent was to allow economizer operation without compressor operation. Here, if the compressor is off the code won't make it to economizer operation (unless I am not interpreting `CoolSpeedNum == 0` correctly, which as you now know, is easy to do).

                if (CoolSpeedNum == 0) {
                } else if (this->m_EconoSpeedNum > 0) { // multi-stage economizer operation
                } else if (CoolSpeedNum > 0) {

I recall the plots and thought it looked like everything was working correctly, but looking at this I wonder if that's true for both choices. It looks like this would work for InterlockedWithMechanicalCooling, but for EconomizerFirst how could the economizer ramp up when the compressor was off? It looks like the first 2 stacks should be switched (with added conditionals), but remember, I'm looking at this on git and not in the debugger.

    A20; \field Economizer Operation Staging
       \type choice
       \key EconomizerFirst
       \key InterlockedWithMechanicalCooling

Reply:
Yes, this is confusing but `m_CoolingSpeedNum` (and hence `CoolSpeedNum`) is at least `> 1` if `m_EconoSpeedNum` is `> 1`, see the snippet below:

```
        this->m_EconoSpeedNum = 0;
        if (this->OAControllerEconomizerStagingType == DataHVACGlobals::EconomizerStagingType::EconomizerFirst) {
            manageEconomizerStagingOperation(state, AirLoopNum, FirstHVACIteration, ZoneLoad);
        }

        // if a variable speed unit, the SensOutputOff at SpeedNum=1 must be checked to see if it exceeds the ZoneLoad
        // This is still no load but at the first speed above idle
        if ((state.dataUnitarySystems->HeatingLoad && this->m_NumOfSpeedHeating > 0) ||
            (state.dataUnitarySystems->CoolingLoad && this->m_NumOfSpeedCooling > 0)) {
            if (this->m_Staged) {
                if (state.dataUnitarySystems->HeatingLoad) {
                    this->m_HeatingSpeedNum = this->m_StageNum;
                } else {
                    this->m_CoolingSpeedNum = std::abs(this->m_StageNum);
                }
            } else {
                if (state.dataUnitarySystems->HeatingLoad) {
                    this->m_HeatingSpeedNum = 1;
                } else {
                    this->m_CoolingSpeedNum = 1;
                }
            }
            // calcUnitarySystemToLoad calls setOnOffMassFlowRate so probably no need to call this here
            this->setOnOffMassFlowRate(state, OnOffAirFlowRatio, PartLoadRatio);
```